lightly/api/api_workflow_tags.py

```diff
@@ -73,3 +73,75 @@ def get_filenames_in_tag(
             masked_select_from_list(filenames_on_server)
 
         return filenames_tag
+
+    def create_tag_from_filenames(
+        self,
+        fnames_new_tag: List[str],
+        new_tag_name: str,
+        parent_tag_id: str = None
+    ) -> TagData:
+        """Creates a new tag from a list of filenames.
+
+        Args:
+            fnames_new_tag:
+                A list of filenames to be included in the new tag.
+            new_tag_name:
+                The name of the new tag.
+            parent_tag_id:
+                The tag defining where to sample from, default: None resolves to the initial-tag.
+
+        Returns:
+            The newly created tag.
+
+        Raises:
+            RuntimeError
+        """
+        
+        # make sure the tag name does not exist yet
+        tags = self.get_all_tags()
+        if new_tag_name in [tag.name for tag in tags]:
+            raise RuntimeError(f'There already exists a tag with tag_name {new_tag_name}.')
+        if len(tags) == 0:
+            raise RuntimeError('There exists no initial-tag for this dataset.')
+
+        # fallback to initial tag if no parent tag is provided
+        if parent_tag_id is None:
+            parent_tag_id = tags[-1].id
+
+        tot_size = tags[-1].tot_size
+
+        # get list of filenames from tag
+        fnames_server = self.get_filenames()
+
+        # create new bitmask for the new tag
+        bitmask = BitMask(0)
+        fnames_new_tag = set(fnames_new_tag)
+        for i, fname in enumerate(fnames_server):
+            if fname in fnames_new_tag:
+                bitmask.set_kth_bit(i)
+        
+        # quick sanity check
+        num_selected_samples = len(bitmask.to_indices())
+        if num_selected_samples != len(fnames_new_tag):
+            raise RuntimeError(
+                f'An error occured when creating the new subset! '
+                f'Found {num_selected_samples} samples newly selected '
```

Comment:
`"newly selected"` is weird. What/who did a selection of what from what?

I would rather say:
f"Out of the {len(fnames_new_tag)} filenames you provided to create a new tag, only {num_selected_samples} have been found on the server. Make sure you use the correct filenames. Valid filename example from the dataset: {fnames_server[0]}'"